content-security-policy/script-src/script-src-strict_dynamic_discard_whitelist.html

```diff
@@ -20,8 +20,9 @@
                     assert_unreached('Whitelisted script without a correct nonce is not allowed with `strict-dynamic`.');
                 }
             }));
-            window.addEventListener('securitypolicyviolation', t.step_func_done(function(e) {
+            window.addEventListener('securitypolicyviolation', t.step_func(function(e) {
                 assert_equals(e.violatedDirective.split(' ')[0], 'script-src');
+                t.done();
```

Comment:
Nit: This change doesn't seem necessary. You're not exiting the handler early in any case; there's only one event, and when it fires, you're done. `step_func_done` seems like the right thing to use here.